pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/juniperkits/JuniperKitCreationStatusTest.java

```diff
@@ -66,68 +87,109 @@ public static void deleteJuniperInstance() {
+        when(mockParticipantLabel.getLabelUrl()).thenReturn("MOCK_LABEL_URL");
+        when(mockEasyPostAddress.getId()).thenReturn("SOME_STRING");
+        when(mockEasyPostUtil.getEasyPostAddressId(any(), any(), any())).thenReturn("SOME_STRING");
+        try {
+            when(mockEasyPostUtil.createParcel(any(), any(), any(), any())).thenReturn(mockEasyPostParcel);
+            when(mockEasyPostUtil.getAddress(any())).thenReturn(mockEasyPostAddress);
+            when(mockEasyPostUtil.createAddressWithoutValidation(any(), any(), any(), any(), any(), any(), any(), any())).thenReturn(
+                    mockEasyPostAddress);
+            when(mockEasyPostUtil.buyShipment(any(), any(), any(), any(), any(), any(), any(), any())).thenReturn(mockEasyPostShipment);
+        } catch (EasyPostException e) {
+            throw new RuntimeException(e);
+        }
+        List<KitRequestShipping> oldKits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", salivaKitType);
+        KitResponse kitResponse =
+                nonPepperKitCreationService.createNonPepperKit(juniperTestKit, salivaKitType, mockEasyPostUtil, ddpInstance);
         Assert.assertFalse(kitResponse.isError());
         Assert.assertNotNull(kitResponse.getKits());
         Assert.assertEquals(1, kitResponse.getKits().size());
-        Assert.assertEquals(juniperTestKit.getJuniperKitId(), kitResponse.getKits().get(0).getJuniperKitId());
-        Assert.assertNotNull(kitResponse.getKits().get(0).getDsmShippingLabel());
+        NonPepperKitStatusDto kitStatus = kitResponse.getKits().get(0);
+        Assert.assertEquals(juniperTestKit.getJuniperKitId(), kitStatus.getJuniperKitId());
+        Assert.assertNotNull(kitStatus.getDsmShippingLabel());
+        Assert.assertNotNull(kitStatus.getCollaboratorSampleId());
+        Assert.assertNotNull(kitStatus.getCollaboratorParticipantId());
         List<KitRequestShipping> newKits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", salivaKitType);
-        Assert.assertEquals(oldkits.size() + 1, newKits.size());
+        Assert.assertEquals(oldKits.size() + 1, newKits.size());
+        KitRequestShipping newKit =
+                newKits.stream().filter(kitRequestShipping -> kitRequestShipping.getDdpParticipantId()
+                                .equals(juniperTestKit.getJuniperParticipantID()))
+                        .findAny().get();
+        Assert.assertEquals(newKit.getBspCollaboratorParticipantId(), bspPrefix + "_" + juniperTestKit.getJuniperParticipantID());
+        Assert.assertEquals(newKit.getBspCollaboratorSampleId(),
+                bspPrefix + "_" + juniperTestKit.getJuniperParticipantID() + "_" + salivaKitType);
     }
 
-    private void assertStatusKitResponse(KitResponse kitResponse, JuniperKitRequest juniperTestKit, int testRandomNumber) {
+    private NonPepperKitStatusDto assertStatusKitResponse(KitResponse kitResponse, JuniperKitRequest juniperTestKit, int testRandomNumber,
```

Comment:
testRandomNumber is generated everytime with `int rand = new Random().nextInt() & Integer.MAX_VALUE;` and then passed into this. it makes finding newly created kits easier by their id

Reply:
Right, I get that. I was simply suggesting constructing the kit ID and passing that around instead, since this 'random' number is really just part of an ID at this point, which you reconstruct in this method. Just makes things easier to follow if you pass around the actual ID. As I said, just a suggestion, not necessary for this review.